public/app/features/explore/RichHistory/RichHistoryCard.tsx

```diff
@@ -147,28 +151,37 @@ export function RichHistoryCard(props: Props) {
   } = props;
   const [activeUpdateComment, setActiveUpdateComment] = useState(false);
   const [comment, setComment] = useState<string | undefined>(query.comment);
+  const [queryDsInstance, setQueryDsInstance] = useState<DataSourceApi | undefined>(undefined);
+
+  useEffect(() => {
+    getQueryDsInstance();
+  });
```

Comment:
This feels a bit wrong. Do we really want to run this on every render? Seems like this should actually create a infinite render loop. Maybe we should just run it if `query.datasourceName` changes.

Reply:
Yeah, good catch! It should run only on first render, as `query.datasourceName` never changes. I'll add `,[]`.